typescript/looker/rtl/nodeSettings.ts

```diff
@@ -68,26 +75,49 @@ export const ApiConfigSection = (
 }
 
 /**
- * .ini Configuration initializer
- * @class NodeSettingsIni
+ * Configuration initializer
+ * @class NodeSettings
  *
  */
-export class NodeSettingsIni extends ApiSettings {
+export class NodeSettings extends ApiSettings {
   constructor(contents: string, section?: string) {
     const settings = ApiConfigSection(contents, section)
     super(settings)
   }
 }
 
+const readEnvConfig = () => {
+  const defaults = DefaultSettings()
+  return {
+    [strLookerApiVersion]: process.env[strLookerApiVersion] || defaults.api_version,
+    [strLookerBaseUrl]: process.env[strLookerBaseUrl] || defaults.base_url,
+    [strLookerVerifySsl]: process.env[strLookerVerifySsl] || defaults.verify_ssl.toString(),
+    [strLookerTimeout]: process.env[strLookerTimeout] || defaults.timeout.toString(),
+    [strLookerClientId]: process.env[strLookerClientId] || '',
+    [strLookerClientSecret]: process.env[strLookerClientSecret] || '',
+  }
+}
+
+export class NodeSettingsEnv extends ApiSettings {
+  constructor() {
+    const settings = ValueSettings(readEnvConfig())
+    super(settings)
+  }
+
+  // @ts-ignore
+  readConfig(section?: string) {
+    return readEnvConfig()
+  }
+}
 /**
- * Example class that reads INI configuration from a file in node
+ * Example class that reads a configuration from a file in node
  * @class NodeSettingsIniFile
  *
  * Warning: INI files storing credentials should be secured in the run-time environment, and
  * ignored by version control systems so credentials never get checked in to source code repositories
  */
-export class NodeSettingsIniFile extends NodeSettingsIni
-  implements IApiSettingsIniFile {
+export class NodeSettingsIniFile extends NodeSettings
```

Comment:
are there other subclassers of `NodeSettings`? If not, maybe combine `NodeSettingsIniFile` and `NodeSettings`?

Reply:
Hmm. I thought I renamed `IApiSettingsIniFile` to `IApiSettingsConfig`. And there's also `NodeSettingsEnv` now. I'll go over with you in person before I land the final version of this branch. I mainly wanted to get the Node env beta testing customer unblocked but should still review this again. Thanks for the close examination.